python/cudf/cudf/core/dataframe.py

```diff
@@ -5866,8 +5866,18 @@ def to_struct(self, name=None):
         -----
         Note that a copy of the columns is made.
         """
+        if not all(isinstance(name, str) for name in self._data.names):
+            warnings.warn(
+                "DataFrame contains non-string column name(s). Struct column "
+                "requires field name to be string. Non-string column names will "
+                "be casted to string as the field name."
+            )
+            field_names = [str(name) for name in self._data.names]
+        else:
+            field_names = self._data.names
```

Comment:
It's probably okay to just always call `str()` here after issuing the warning for non-string types. Minimizes the logic -- and avoids potential issues with `self._data.names` returning a tuple, while the other path uses a list. We saw another instance of that recently in a hotfix for 22.02. 😉
```suggestion
        field_names = [str(name) for name in self._data.names]
```